TypeProvider/Core/Prelude.fs

```diff
@@ -0,0 +1,22 @@
+[<AutoOpen>]
+module Froto.TypeProvider.Core.Prelude
+
+open System
+open System.IO
+
+open Printf
+
+/// Raises NotSupportedException with given string as message
+let notsupportedf fmt = ksprintf (NotSupportedException >> raise) fmt
```

Comment:
Should Froto have a hierarchy of exceptions, so a caller can trap generic parser, serializer, or type provider errors with a single type?